### PR TITLE
Remove Surface_mesher demo (was "Fix Surface_mesher_demo Apple warnings")

### PR DESCRIPTION
**Pull request by danston**
Apple deprecated OpenGL starting from macOS 10.14 and so the `Surface_mesher` demo shows a bunch of related warnings when tested on the corresponding macOS platforms. I propose this little PR to suppress these warnings.

EDIT: Removes the Surface_mesh_Demo completely.

## Release Management

* Affected package(s): Surface_mesher
* Issue(s) solved (if any): warnings
* Feature/Small Feature (if any): bug fix
* License and copyright ownership: no changes

### CODE REVIEW

**Comment by afabri** [comment]
@lrineau @maxGimeno what do we have in the surface mesher demo that we have not in the polyhedron demo?

**Comment by maxGimeno** [comment]
AFAIK nothing.

**Comment by afabri** [comment]
You probably still have to change in the PkgDescription.txt file that the demo is now in Polyhedron.

**Comment by maxGimeno** [comment]
done

File: Surface_mesher/doc/Surface_mesher/PackageDescription.txt [hunk 1]
```diff
@@ -38,7 +38,7 @@
 \cgalPkgDependsOn{\ref PkgTriangulation3}
 \cgalPkgBib{cgal:ry-smg}
 \cgalPkgLicense{\ref licensesGPL}
-\cgalPkgDemo{Surface Mesh Generator,surface_mesher.zip}
+\cgalPkgDemo{Polyhedron demo,polyhedron_3.zip}
```
Comment (lrineau) [inline Surface_mesher/doc/Surface_mesher/PackageDescription.txt:41]:
```suggestion
```

The polyhedron demo does not use `Surface_mesher` at all (grepped for it). It uses Mesh_3 to implement similar functions.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-78.shtml

**Pull request merged**